### ISSUE TITLE
[Vita] Pokemon LeafGreen not saving (intermittent)

### ISSUE THREAD

**Issue by fqborges** [mgba-emu/mgba#340]
Symptoms: 
- Played a while, saved and quit mgba, game **is** saved.
- Played a while, saved and quit mgba, game is **not** saved.
- Played a while, saved and quit game (not mgba) to check, game **is** saved.
- Played a while, saved multiple times and quit game. Checked again, and game is **not** saved.

Version: yesterday nightly build (2016-08-11).

Hint: I cannot test this now, but maybe opening configs ( /_\ button) makes difference.

**Comment by endrift** [comment]
Can you let me know if this build fixes the issue? http://endrift.com/files/mgba-savesync.vpk

**Comment by fqborges** [comment]
Unfortunately, It didn't.

Using the provided buid I started a new game, saved, waited a few seconds and left the emulator. After restarting, the game I just saved wasn't there.

I am going to test more, and provide you more feedback. For now, this problem can be bypassed using saved states at least.

I really appreciate your work. Thanks!

**Comment by endrift** [comment]
How did you leave the emulator? Did you exit the game and go down through the menu, or did you tear off the livearea page?

**Comment by fqborges** [comment]
I did tear off the livearea page.

**Comment by endrift** [comment]
This may be fixed in the upcoming nightly. I've had trouble reproducing the issue properly, so let me know.

**Comment by endrift** [comment]
I've figured out what's going on:

Pokémon games on the GBA have two savedata blocks, and it swaps off on which one it writes to. It also keeps track of how many times you've saved, and the savegame that says "we've saved the most times" is the one that's loaded next time.

So what happens if you take a savestate, save three times, load the savestate, and save once? Well the save file has two games, one with N saves and one with N+2. However, the N+2 one is older, from before reloading the savestate and saving again. So it loads the wrong one.

The way to fix this would be load back in the savedata when loading a savestate, and only flush it if a savegame is written out.
